src/limb.rs

```diff
@@ -0,0 +1,271 @@
+        for _ in 0..100 {
+
+            // NSA Guide Step 4.
+            //
+            // The requirement that the random number generator has the requested
+            // security strength is delegated to `rng`.
+            {
+                let mut dest_as_bytes = limbs_as_bytes_mut(dest);
+                try!(rng.fill(&mut dest_as_bytes));
+            }
+
+            let bits_to_mask = self.leading_zero_bits();
+            if bits_to_mask > 0 {
+                let keep: Limb = (1 << (LIMB_BITS - bits_to_mask)) - 1;
+                dest[self.limbs.len() - 1] &= keep;
+            }
+
+            // NSA Guide Steps 5, 6, and 7.
+            //
+            // XXX: The NSA guide says that we should verify that the random scalar
+            // is in the range [0, n - 1) and then add one to it so that it is in
+            // the range [1, n). Instead, we verify that the scalar is in the range
+            // [1, n) like BoringSSL (et al.) does. This way, we avoid needing to
+            // compute or store the value (n - 1), we avoid the need to implement
+            // a function to add one to a scalar, and we avoid needing to convert
+            // the scalar back into an array of bytes. TODO: Is there any security
+            // advantage to the way the NSA suggests? There doesn't seem to be,
+            // other than being less error prone w.r.t. accidentally generating
+            // zero-valued keys.
+            if !self.are_limbs_within(&dest) {
+                continue;
+            }
+
+            // NSA Guide Step 8 is done in `private_to_public()`.
+
+            // NSA Guide Step 9.
+            return Ok(());
+        }
+
+        Err(())
+    }
+
+}
+
+#[allow(unsafe_code)]
+#[allow(non_snake_case)]
+#[doc(hidden)]
+#[no_mangle]
+pub unsafe extern fn GFp_rand_mod(dest: *mut Limb, range: *const Limb,
+                                  num_limbs: c::size_t, rng: *mut rand::RAND)
+                                  -> c::int {
+    const ERR: c::int = 0;
+    const SUCCESS: c::int = 1;
+
+    let range = Range::from_limbs(core::slice::from_raw_parts(range, num_limbs));
+    if !range.is_ok() {
+        return ERR;
+    }
+
+    let mut dest = core::slice::from_raw_parts_mut(dest, num_limbs);
+
+    let result = range.unwrap().sample_into_limbs(&mut dest, (*rng).rng);
+    if result.is_err() {
+        ERR
+    } else {
+        SUCCESS
+    }
+}
+
+
+#[cfg(target_pointer_width = "64")]
+fn limbs_as_bytes_mut<'a>(src: &'a mut [Limb]) -> &'a mut [u8] {
+    polyfill::slice::u64_as_u8_mut(src)
+}
+
+#[cfg(target_pointer_width = "32")]
+fn limbs_as_bytes_mut<'a>(src: &'a mut [Limb]) -> &'a mut [u8] {
+    polyfill::slice::u32_as_u8_mut(src)
+}
+
+#[allow(unsafe_code)]
+pub fn limbs_less_than_limbs_constant_time(a: &[Limb], b: &[Limb]) -> bool {
+    assert_eq!(a.len(), b.len());
+    let result = unsafe {
+        GFp_constant_time_limbs_lt_limbs(a.as_ptr(),
+                                         b.as_ptr(),
```

Comment:
I guess this is the result of rustfmt or following rustfmt style. But, I use the BoringSSL style, which is to wrap to as few lines as possible. Please use the BoringSSL style. (If you know a way to write a rustfmt configuration file that does that, I'm happy to add one to this repo to make it easier for people..)